### PR TITLE
Add NC name pill to Search and Filters modal on address selection

### LINKED ISSUE #1813: Add NC name pill to Search and Filters modal on address selection
### Overview

We need to render the relevant neighborhood council's name as a pill when the user selects an address so that users know what neighborhood council that address is in.

### Action Items

- [ ] Ensure that address search also selects the NC as a pill (StyledChip)
  - [ ] Capture councilId param value when Address is searched
  - [ ] Display NC in a StyledChip with a `x` close action
- [ ] Ensure that clicking `x` on the pill will deselect the selected address and reset the zoom level to see all boundaries (e.g. the default zoom level)
  - [ ] Do this by using the `this.reset` function in the resetMap prop  

### Resources/Instructions

Useful Files and Components
- Map: `components/Map/Map.jsx`
  - see: `onGeocoderResult`
  - see: resetMap prop in `<FilterMenu/>`

v1.2 Moderated Usability Test Slideshow:
- [link](https://docs.google.com/presentation/d/1w1YpzC-sM7X1OOQ70f5Ey8h6Q0yOoTlC/edit)
- [Slide for Insight (1)](https://docs.google.com/presentation/d/1w1YpzC-sM7X1OOQ70f5Ey8h6Q0yOoTlC/edit#slide=id.p9)

<details><summary>Screenshot before proposed changes</summary>
<p>

image

</p>
</details>

<details><summary>Screenshot after proposed changes</summary>
<p>
image
</p>
</details>

### PR DESCRIPTION
**Pull request by Brandoncyu**
Fixes #1813 

  - [x] Up to date with `main` branch
  - [x] Branch name follows [guidelines](https://github.com/hackforla/311-data/blob/master/GETTING_STARTED.md#feature-branching)
  - [x] All PR Status checks are successful
  - [x] Peer reviewed and approved

Any questions? See the [getting started guide](https://github.com/hackforla/311-data/blob/master/GETTING_STARTED.md)

### CODE REVIEW

**Comment by traycn** [comment]
Review ETA: Fri 9/13 PT at 12PM

File: components/Map/Map.jsx
Comment (ryanfchase) [inline components/Map/Map.jsx:566]: Let's add an error handler after this `find` lookup.
```js
if (!newSelectedCouncil) {
  throw new Error('Council Id in address search geocoder result could not be found');
}
```

**Review by ryanfchase — CHANGES_REQUESTED** [review]
Looks good. Just wanted to add some error handling, but otherwise this is looking consistent with similar nc selection behaviors across the application

File: components/Map/Map.jsx
Comment (Brandoncyu) [reply components/Map/Map.jsx:566]: Done

**Review by ryanfchase — APPROVED** [review]
LGMT 👍

**Comment by ryanfchase** [comment]
Just left my approval. Let’s wait for @traycn’s review before merging. Thanks.

**Comment by traycn** [comment]
Screen Shot 2024-09-13 at 12 09 46 PM

Pulled locally and works as expected. Great solution! LGTM for merge!

**Review by traycn — APPROVED** [review]

**Comment by Brandoncyu** [comment]
Merged!

**Pull request merged**
